src/BuildScriptGeneratorCli/Commands/ExecCommand.cs

```diff
@@ -0,0 +1,104 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using McMaster.Extensions.CommandLineUtils;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.BuildScriptGenerator;
+using Microsoft.Oryx.Common;
+using Microsoft.Oryx.Common.Extensions;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli
+{
+    [Command(Name, Description = "Execute an arbitrary command in the default shell, in an environment " +
+        "with the best-matching platform tool versions.")]
+    internal class ExecCommand : CommandBase
+    {
+        public const string Name = "exec";
+
+        public const string DefaultBashPath = "/bin/bash";
+
+        [Argument(0, Description = "The source directory.")]
+        public string SourceDir { get; set; }
+
+        [Argument(1, Description = "The command to execute in an app-specific environment.")]
+        public string Command { get; set; }
+
+        internal override int Execute(IServiceProvider serviceProvider, IConsole console)
+        {
+            var logger = serviceProvider.GetRequiredService<ILogger<BuildCommand>>();
```

Comment:
`ILogger<ExecCommand>`

Reply:
yeah, fixed already but did not push. thanks!